torch/lib/c10d/test/ProcessGroupMPITest.cpp

```diff
@@ -36,13 +41,14 @@ void testAllreduce(int iter = 1000) {
 
   std::vector<c10::intrusive_ptr<::c10d::ProcessGroup::Work>> works;
   for (auto& tensors : allTensors) {
```

Comment:
The old code needs `allTensor` because it needs to verify outputs using `allTensor`. Now the new code get output directly from `waitWork()`, why do we still need `allTensors`? Will it work if we just construct tensors and pass them to `pg->allreduce` in this loop? Or is it because the application code should keep the tensors alive and so that temporary tensors won't work?